src/pages/Produtos.js

```diff
@@ -2,13 +2,37 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 class Produtos extends React.Component {
-  renderProduto = ({ id, title, thumbnail, price }) => (
-    <li key={ id } data-testid="product">
-      <p>{ title }</p>
-      <img src={ thumbnail } alt={ title } />
-      <p>{ price }</p>
-    </li>
-  );
+  constructor() {
+    super();
+
+    this.state = {
+      carrinho: [],
+    };
+  }
+
+  addAoCarrinho = (produto) => {
+    this.setState((state) => ({ carrinho: [...state.carrinho, produto] }));
```

Comment:
```suggestion
    this.setState(({carrinho}) => ({ carrinho: [...carrinho, produto] }));
```
Opa Diego, seria possível desestruturar no setState o estado, assim deixando o codigo mais enxuto, concordas?